src/frontends/tensorflow_common/src/op/select.cpp

```diff
@@ -40,10 +52,25 @@ OutputVector translate_select_v2_op(const NodeContext& node) {
     // is true or the value of 'y' if false. There are valid condition input sizes:
     // 1. Either the same shape (in which case the select is elementwise), or
     // 2. Broadcastable shapes between 'condition', 'x' and 'y'.
-    default_op_checks(node, 3, {"SelectV2", "SELECT_V2"});
-    // no preparation for inputs are needed
-    // inputs are already NumPy broadcastable
-    return translate_select_base_op(node, node.get_input(0), node.get_input(1), node.get_input(2));
+    default_op_checks(node, 3, {"SelectV2", "SELECT_V2"}, true);
+    auto condition = node.get_input(0);
+    auto x = node.get_input(1);
+    auto y = node.get_input(2);
+
+    element::Type complex_part_type;
+    auto is_complex = has_complex_inputs(x, y, complex_part_type);
+
+    if (is_complex) {
+        auto cur_cond_shape = make_shared<v3::ShapeOf>(condition, element::i32);
+        auto const_one = make_shared<v0::Constant>(element::i32, Shape{1}, 1);
+        auto new_cond_shape = make_shared<v0::Concat>(OutputVector{cur_cond_shape, const_one}, 0);
+        auto new_condition = make_shared<v1::Reshape>(condition, new_cond_shape, false);
```

Comment:
please use Unsqueeze by `-1` dimension instead. It should be simpler.

Reply:
done